hadoop-yarn-project/hadoop-yarn/hadoop-yarn-api/pom.xml

```diff
@@ -55,6 +55,7 @@
       <groupId>org.apache.hadoop</groupId>
       <artifactId>hadoop-annotations</artifactId>
     </dependency>
+
```

Comment:
```suggestion
```

Reply:
> I guess the deprecation warnings should be handled in a follow-up patch as you mentioned in the conversation in the jira.

Since all warnings are in generated code and still we want to use "proto2" syntax for generated code, I guess these will stay.


>change in hadoop-yarn-api
This is intentional change to include hadoop-yarn-api during jenkins test.
 During patch compilation "hadoop-yarn-common" was referring to hadoop-yarn-api, and without this hadoop-yarn-api will have old generated code. Because of which hadoop-yarn-common compilation was failing.

Reply:
Is the complaint about warning messages because proto files are in proto2 format? Does this help? https://issues.apache.org/jira/browse/HBASE-18866

Reply:
No, this is not about that. This is about javac warnings in generated java code. Mostly due to (default) "proto2" syntax in proto files.

I have handled deprecated APIs in non-generated code.